orca-clouddriver/src/main/java/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/MonitorDeployManifestTask.java

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright 2021 Salesforce, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.manifest;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.kork.annotations.VisibleForTesting;
+import com.netflix.spinnaker.kork.core.RetrySupport;
+import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService;
+import com.netflix.spinnaker.kork.retrofit.exceptions.SpinnakerHttpException;
+import com.netflix.spinnaker.kork.retrofit.exceptions.SpinnakerServerException;
+import com.netflix.spinnaker.orca.api.pipeline.models.StageExecution;
+import com.netflix.spinnaker.orca.clouddriver.KatoService;
+import com.netflix.spinnaker.orca.clouddriver.OortService;
+import com.netflix.spinnaker.orca.clouddriver.model.Manifest;
+import com.netflix.spinnaker.orca.clouddriver.model.Task;
+import com.netflix.spinnaker.orca.clouddriver.model.TaskOwner;
+import com.netflix.spinnaker.orca.clouddriver.tasks.MonitorKatoTask;
+import com.netflix.spinnaker.orca.pipeline.model.PipelineExecutionImpl;
+import com.netflix.spinnaker.orca.pipeline.model.SystemNotification;
+import groovy.util.logging.Slf4j;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class MonitorDeployManifestTask extends MonitorKatoTask {
```

Comment:
One thing to note, is it seems like this task is only possible if dynamic configuration is used?

If that is the case, can we add comments or some documentation outlining that?

Reply:
I don't really like the DynamicConfigService, but I think it's always available...or at least orca wouldn't start without it, even before this PR.

Reply:
It always available, but this line https://github.com/spinnaker/orca/pull/4644/files/1984caaf0cb4ed7e5dd282f1687202d19e55c74f#diff-2ba79f1fbda91e9376143202ef17da92baa1bae4f6c520e8146360de93bf7c64R104 will always return false. Since the default dynamic configuration just return the default value. I think that means this feature will **only** work if it is present and not returning false?

Reply:
The whole term in the `if` expression is
```
!getDynamicConfigService()
            .isEnabled("tasks.monitor-kato-task.kubernetes.deploy-manifest.retry-task", false))
```
so yes, this feature only works when enabled.

Reply:
Okay, can we call that out in documentation somewhere? I know it should be obvious when looking at the code, but may be a quick and easy determination if it is documented at the top of the class.

Mostly cause I was like "Oooo, can we use this feature?" But we dont use dynamic configuration ><

Reply:
Hmmm...I feel like the dynamic config bits are an implementation detail.  The PR description does have:
> The following configuration flags (and their defaults) control the behavior of this new feature:
```
tasks.monitor-kato-task.kubernetes.deploy-manifest:
  retry-task.enabled: false
  maximum-period-inactivity-ms: 300000L
  maximum-forced-retries: 3
  account: "" # the name of the account where clouddriver runs, required if enabled is true
  namespace: spinnaker
```
and I figured I'd add the same to the release notes.  Is that enough?

Reply:
Yep! Works for me